includes/multi-currency/MultiCurrency.php

```diff
@@ -616,6 +618,32 @@ private function cache_currencies( $currencies, int $updated = null, int $expira
 		return $result;
 	}
 
+	/**
+	 * Checks to see if the store currency has changed and puts any manual rate currencies into an option for a notice to display.
+	 */
+	private function check_store_currency_for_change() {
+		$last_known_currency  = get_option( $this->id . '_store_currency', false );
+		$woocommerce_currency = get_woocommerce_currency();
+		if ( $last_known_currency && $last_known_currency !== $woocommerce_currency ) {
+			$enabled_currencies = $this->get_enabled_currencies();
+			$manual_currencies  = [];
+
+			// Check enabled currencies for manual rates.
+			foreach ( $enabled_currencies as $currency ) {
+				$rate_type = get_option( $this->id . '_exchange_rate_' . $currency->get_id(), false );
+				if ( 'manual' === $rate_type ) {
+					$manual_currencies[] = $currency->get_name();
+				}
+			}
+
+			if ( 0 < count( $manual_currencies ) ) {
+				update_option( $this->id . '_show_store_currency_changed_notice', $manual_currencies );
+			}
+		}
+
+		update_option( $this->id . '_store_currency', $woocommerce_currency );
```

Comment:
If the `$last_known_currency` is the same as `$woocommerce_currency` we could consider skipping this call to `update_option`? 

On the other hand, I think `update_option` doesn't update if the current option value is the same as the new option value, so maybe this is a micro-optimisation.

Reply:
I checked and `update_option` doesn't update if the values match, however, it does call `get_option` again, so if the value isn't already in the cache (which it should be), then another db query is made. 

There are three outcomes here:
- If `$last_known_currency` is false.
- if `$last_known_currency !== $woocommerce_currency`.
- If `$last_known_currency === $woocommerce_currency`.

For the first two, we need to update the option, for the last we don't. This means we can do something like the below, or a nested `if`, but in either case, I think the update call would need to happen twice:

```
	private function check_store_currency_for_change() {
		$last_known_currency  = get_option( $this->id . '_store_currency', false );
		$woocommerce_currency = get_woocommerce_currency();
		if ( ! $last_known_currency ) {
			update_option( $this->id . '_store_currency', $woocommerce_currency );
			return;
		}

		if ( $last_known_currency !== $woocommerce_currency ) {
			$enabled_currencies = $this->get_enabled_currencies();
			$manual_currencies  = [];

			// Check enabled currencies for manual rates.
			foreach ( $enabled_currencies as $currency ) {
				$rate_type = get_option( $this->id . '_exchange_rate_' . $currency->get_id(), false );
				if ( 'manual' === $rate_type ) {
					$manual_currencies[] = $currency->get_name();
				}
			}

			if ( 0 < count( $manual_currencies ) ) {
				update_option( $this->id . '_show_store_currency_changed_notice', $manual_currencies );
			}
			update_option( $this->id . '_store_currency', $woocommerce_currency );
		}
	}
```

Reply:
I think that's a good solution, it updates the option and exits the function early if we didn't have a `$last_known_currency`, and in any other case it will only update the option if the last known currency is not the same as the store currency. 

I don't think it's an issue having two calls to `update_option` there twice, because in practice they would never both be called.

Reply:
Okay, makes sense. I just wasn't sure if we should try not to add the same update call twice or not. I updated to use the code above in https://github.com/Automattic/woocommerce-payments/pull/2322/commits/b6ccb7d5165686c2bc4fdee0932a24f949d8fdfa

Reply:
Awesome, thanks for making that change! I think generally avoiding having the same update statement twice is good, but in this case I can't really see another way to do it, and as you have a `return;` statement right after one of them, I think it's pretty clear that we'll only ever call one (or none) of the `update_option`'s in a single execution. 🙂